### PR TITLE
[REVIEW] Adding templates for Issues and Pull Requests

### PR DESCRIPTION
**Pull request by mike-wendt**
These are the same templates used by PyGDF with a name change to libGDF

Link to the approved/merged PR for PyGDF - https://github.com/gpuopenanalytics/pygdf/pull/179

### CODE REVIEW

**Comment by mike-wendt** [comment]
skip ci

**Comment by mike-wendt** [comment]
gpuCI failure due to hashing test issue, unrelated to changes in PR

File: ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+Before submitting a bug report please ensure that you can check off these boxes:
+
+-->
+
+- [ ] I am using the latest version of libGDF from conda or built from `master`.
+- [ ] I have included the following environment details:
+      Linux Distro, Linux Kernel, GPU Model
+- [ ] I have included the following version information for:
+      Arrow, CUDA, Numpy, Pandas, Python
```
Comment (kkraus14) [inline ISSUE_TEMPLATE.md]: I'm not sure how relevant Numpy / Pandas are here. I think we can replace Numpy and Pandas with GCC version and CMake version as a starting point.

**Comment by kkraus14** [comment]
Looks good to me, thanks @mike-wendt!

**Pull request merged**